### PR TITLE
server: when init data dir, skip if get changefeed info failed.

### LINKED ISSUE #2777: TiCDC scale out failed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
* create changefeed
* remove changefeed
* scale out ticdc

Detailed refer to http://172.16.4.80:31090/view/UTF/job/utf-go-test/2448/console

2. What did you expect to see?
scale out successfully

3. What did you see instead?
scale out failed
image

```
kubectl logs ticdc1-srhavths-ticdc-1 -n jenkins-ci
[WARN] TiCDC server data-dir is not set. Please use `cdc server --data-dir` to start the cdc server if possible.
[2021/09/09 02:40:19.160 +00:00] [INFO] [util.go:99] ["init log"] [file=/dev/stderr] [level=debug]
[2021/09/09 02:40:19.160 +00:00] [INFO] [version.go:47] ["Welcome to Change Data Capture (CDC)"] [release-version=v5.0.4] [git-hash=0ebf717e1fce3ddeed786c5241669cab48761ad1] [git-branch=heads/refs/tags/v5.0.4] [utc-build-time="2021-09-06 12:49:45"] [go-version="go version go1.13 linux/amd64"] [failpoint-build=false]
[2021/09/09 02:40:19.160 +00:00] [INFO] [server.go:74] ["creating CDC server"] [pd-addrs="[http://ticdc1-srhavths-pd:2379]"] [config="{\"addr\":\"0.0.0.0:8301\",\"advertise-addr\":\"ticdc1-srhavths-ticdc-1.ticdc1-srhavths-ticdc-peer.jenkins-ci.svc:8301\",\"log-file\":\"/dev/stderr\",\"log-level\":\"debug\",\"log\":{\"file\":{\"max-size\":300,\"max-days\":0,\"max-backups\":0}},\"data-dir\":\"\",\"gc-ttl\":86400,\"tz\":\"System\",\"capture-session-ttl\":10,\"owner-flush-interval\":200000000,\"processor-flush-interval\":100000000,\"sorter\":{\"num-concurrent-worker\":4,\"chunk-size-limit\":134217728,\"max-memory-percentage\":30,\"max-memory-consumption\":17179869184,\"num-workerpool-goroutine\":16,\"sort-dir\":\"/tmp/sorter\"},\"security\":{\"ca-path\":\"\",\"cert-path\":\"\",\"key-path\":\"\",\"cert-allowed-cn\":null},\"per-table-memory-quota\":20971520,\"kv-client\":{\"worker-concurrent\":8,\"worker-pool-size\":0,\"region-scan-limit\":40}}"]
[2021/09/09 02:40:19.160 +00:00] [INFO] [client.go:214] ["[pd] create pd client with endpoints"] [pd-address="[http://ticdc1-srhavths-pd:2379]"]
[2021/09/09 02:40:19.172 +00:00] [INFO] [base_client.go:334] ["[pd] update member urls"] [old-urls="[http://ticdc1-srhavths-pd:2379]"] [new-urls="[http://ticdc1-srhavths-pd-0.ticdc1-srhavths-pd-peer.jenkins-ci.svc:2379]"]
[2021/09/09 02:40:19.172 +00:00] [INFO] [base_client.go:346] ["[pd] switch leader"] [new-leader=http://ticdc1-srhavths-pd-0.ticdc1-srhavths-pd-peer.jenkins-ci.svc:2379] [old-leader=]
[2021/09/09 02:40:19.175 +00:00] [INFO] [base_client.go:126] ["[pd] init cluster id"] [cluster-id=7005751877393054850]
[2021/09/09 02:40:19.175 +00:00] [INFO] [client.go:238] ["[pd] create tso dispatcher"] [dc-location=global]
[2021/09/09 02:40:19.179 +00:00] [ERROR] [server.go:132] ["run server"] [error="[CDC:ErrChangeFeedNotExists]changefeed not exists, key: /tidb/cdc/changefeed/info/33e585c5-1785-4da7-8d43-ac19accef69d\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/normalize.go:156\ngithub.com/pingcap/ticdc/cdc/kv.CDCEtcdClient.GetChangeFeedInfo\n\tgithub.com/pingcap/ticdc@/cdc/kv/etcd.go:212\ngithub.com/pingcap/ticdc/cdc.(*Server).setUpDataDir\n\tgithub.com/pingcap/ticdc@/cdc/server.go:435\ngithub.com/pingcap/ticdc/cdc.(*Server).initDataDir\n\tgithub.com/pingcap/ticdc@/cdc/server.go:382\ngithub.com/pingcap/ticdc/cdc.(*Server).Run\n\tgithub.com/pingcap/ticdc@/cdc/server.go:146\ngithub.com/pingcap/ticdc/cmd.runEServer\n\tgithub.com/pingcap/ticdc@/cmd/server.go:130\ngithub.com/spf13/cobra.(*Command).execute\n\tgithub.com/spf13/cobra@v1.0.0/command.go:842\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\tgithub.com/spf13/cobra@v1.0.0/command.go:950\ngithub.com/spf13/cobra.(*Command).Execute\n\tgithub.com/spf13/cobra@v1.0.0/command.go:887\ngithub.com/pingcap/ticdc/cmd.Execute\n\tgithub.com/pingcap/ticdc@/cmd/root.go:32\nmain.main\n\tgithub.com/pingcap/ticdc@/cmd/cdc/main.go:22\nruntime.main\n\truntime/proc.go:203\nruntime.goexit\n\truntime/asm_amd64.s:1357"]
[2021/09/09 02:40:19.179 +00:00] [INFO] [client.go:510] ["[pd] exit tso dispatcher"] [dc-location=global]
Error: run server: [CDC:ErrChangeFeedNotExists]changefeed not exists, key: /tidb/cdc/changefeed/info/33e585c5-1785-4da7-8d43-ac19accef69d
Usage:
  cdc server [flags]

Flags:
      --addr string                           Set the listening address (default "127.0.0.1:8300")
      --advertise-addr string                 Set the advertise listening address for client communication
      --ca string                             CA certificate path for TLS connection
      --cert string                           Certificate path for TLS connection
      --cert-allowed-cn string                Verify caller's identity (cert Common Name). Use ',' to separate multiple CN
      --config string                         Path of the configuration file
      --data-dir string                       the path to the directory used to store TiCDC-generated data
      --gc-ttl int                            CDC GC safepoint TTL duration, specified in seconds (default 86400)
  -h, --help                                  help for server
      --key string                            Private key path for TLS connection
      --log-file string                       log file path
      --log-level string                      log level (etc: debug|info|warn|error) (default "info")
      --owner-flush-interval duration         owner flushes changefeed status interval (default 200ms)
      --pd string                             Set the PD endpoints to use. Use ',' to separate multiple PDs (default "http://127.0.0.1:2379")
      --processor-flush-interval duration     processor flushes task status interval (default 100ms)
      --sorter-chunk-size-limit uint          size of heaps for sorting (default 134217728)
      --sorter-max-memory-consumption uint    maximum memory consumption of in-memory sort (default 17179869184)
      --sorter-max-memory-percentage int      system memory usage threshold for forcing in-disk sort (default 30)
      --sorter-num-concurrent-worker int      sorter concurrency level (default 4)
      --sorter-num-workerpool-goroutine int   sorter workerpool size (default 16)
      --tz string                             Specify time zone of TiCDC cluster (default "System")

run server: [CDC:ErrChangeFeedNotExists]changefeed not exists, key: /tidb/cdc/changefeed/info/33e585c5-1785-4da7-8d43-ac19accef69d
```

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```
TYPE	INSTANCE	STATUS_ADDRESS	VERSION	GIT_HASH	START_TIME	UPTIME	SERVER_ID
tidb	ticdc1-srhavths-tidb-0.ticdc1-srhavths-tidb-peer.jenkins-ci.svc:4000	ticdc1-srhavths-tidb-0.ticdc1-srhavths-tidb-peer.jenkins-ci.svc:10080	5.0.4	53251a9731da02ad9ee5abed9f27a14c7dea33a4	2021-09-09T02:16:13Z	47m33.502694244s	0
pd	ticdc1-srhavths-pd-0.ticdc1-srhavths-pd-peer.jenkins-ci.svc:2379	ticdc1-srhavths-pd-0.ticdc1-srhavths-pd-peer.jenkins-ci.svc:2379	5.0.3	a41aaa7b80f54fb3618d613899ede125e8625f34	2021-09-09T02:15:17Z	48m29.502702697s	0
tikv	ticdc1-srhavths-tikv-0.ticdc1-srhavths-tikv-peer.jenkins-ci.svc:20160	172.18.136.2:20180	5.0.3	84525db9e1e2a7da9ac51f8c9bf7326d8b6537e4	2021-09-09T02:15:43Z	48m3.5027056s	0

```

    - TiCDC version (execute `cdc version`):

        ```
        kubectl exec -it ticdc1-srhavths-ticdc-0 -n jenkins-ci -- /cdc version
        Release Version: v5.0.4
        Git Commit Hash: 0ebf717e1fce3ddeed786c5241669cab48761ad1
        Git Branch: heads/refs/tags/v5.0.4
        UTC Build Time: 2021-09-06 12:49:45
        Go Version: go version go1.13 linux/amd64
        Failpoint Build: false
        ```

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve? 

try to fix #2777

### What is changed and how it works?

skip if find changefeed info failed

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/server.go [hunk 1]
```diff
@@ -300,7 +300,8 @@ func (s *Server) setUpDataDir(ctx context.Context) error {
 	for id := range allStatus {
 		info, err := s.etcdClient.GetChangeFeedInfo(ctx, id)
```
Comment (amyangfei) [inline cdc/server.go:301]: What about using `GetChangeFeeds` to query all changefeed info. (maybe a little change to this function, unmarshall changefeed info inside)

Comment (3AceShowHand) [reply cdc/server.go:301]: we can add a new function `s.etcdClient.GetAllChangeFeedInfo(ctx)`

Comment (amyangfei) [reply cdc/server.go:301]: Fine

File: cdc/kv/etcd.go [hunk 2]
```diff
@@ -201,6 +201,24 @@ func (c CDCEtcdClient) GetChangeFeeds(ctx context.Context) (int64, map[string]*m
 	return revision, details, nil
 }
 
+// GetAllChangeFeedInfo queries all changefeed information
+func (c CDCEtcdClient) GetAllChangeFeedInfo(ctx context.Context) (map[string]*model.ChangeFeedInfo, error) {
```
Comment (amyangfei) [inline cdc/kv/etcd.go:205]: Better to add a unit test for this function

**Comment by liuzix** [comment]
We can launch a local Etcd server and unit test `setUpDataDir`

**Review by liuzix — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**